### PR TITLE
[GSSDLC-127] One Click Deploy now creates two teams: _DEV and _CUSTOMERS

### PR DESCRIPTION
**Pull request by jhonypg**

### CODE REVIEW

File: src/main/java/objective/taskboard/domain/ProjectFilterConfiguration.java [hunk 1]
```diff
@@ -24,30 +26,40 @@
...
+    }
+
+    public void setProjectKey(String projectKey) {
+        this.projectKey = projectKey;
+    }
+
     @OneToMany(fetch=FetchType.EAGER)
-    @JoinColumn(name="projectKey")
+    @JoinColumn(name="projectKey", referencedColumnName="projectKey")
     private List<ProjectTeam> projectTeams;
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/domain/ProjectFilterConfiguration.java]: CRITICAL Make "projectTeams" transient or serializable. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1948)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'

File: src/main/java/objective/taskboard/data/Team.java [hunk 2]
```diff
@@ -68,8 +71,112 @@
...
     private List<UserTeam> members;
+
+    public Team(String name, String manager, String coach, ArrayList<String> members) {
+        this.name = name;
+        this.manager = manager;
+        this.coach = coach;
+        this.members = stringListToUserTeamList(members, name);
+    }
+
+    private List<UserTeam> stringListToUserTeamList(ArrayList<String> members, String teamName) {
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/data/Team.java]: MAJOR Remove this unused method parameter "teamName". [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1172)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: src/main/java/objective/taskboard/data/Team.java [hunk 3]
```diff
@@ -36,19 +40,18 @@
 import javax.persistence.Table;
 
 import lombok.AllArgsConstructor;
-import lombok.Data;
 import lombok.NoArgsConstructor;
 
-@Data
 @Entity
 @NoArgsConstructor
 @AllArgsConstructor
```
Comment (taksan) [inline src/main/java/objective/taskboard/data/Team.java]: Provavelmente dá para remover o @AllArgsConstructor

File: src/main/java/objective/taskboard/controller/ProjectController.java [hunk 4]
```diff
@@ -162,36 +162,58 @@
...
+
+    private void createTeamAndConfigurations(String projectKey, String teamName, String manager, String coach, ArrayList<String> members) {
+        Team team = createTeam(teamName, manager, coach, members);
+        TeamFilterConfiguration teamFilterConfiguration = createTeamFilterConfiguration(team);
+        createProjectTeam(projectKey, teamFilterConfiguration);
+    }
+
+    private Team createTeam(String name, String manager, String coach, ArrayList<String> members) {
+        final Team team = new Team(name, manager, coach, members);
         Team persistedTeam = teamRepository.save(team);
```
Comment (taksan) [inline src/main/java/objective/taskboard/controller/ProjectController.java]: Daria para retornar direto, sem atribuir na variável né?

File: src/main/java/objective/taskboard/controller/ProjectController.java [hunk 5]
```diff
@@ -162,36 +162,58 @@
...
+        return persistedTeam;
+    }
 
+    private TeamFilterConfiguration createTeamFilterConfiguration(Team team) {
         final TeamFilterConfiguration teamFilterConfiguration = new TeamFilterConfiguration();
-        teamFilterConfiguration.setTeamId(persistedTeam.getId());
+        teamFilterConfiguration.setTeamId(team.getId());
         TeamFilterConfiguration persistedTeamFilter = teamFilterConfigurationRepository.save(teamFilterConfiguration);
```
Comment (taksan) [inline src/main/java/objective/taskboard/controller/ProjectController.java]: Daria para retornar sem atribuir

**Pull request merged**